numpy/typing/tests/data/reveal/array_constructors.pyi

```diff
@@ -37,6 +40,10 @@ assert_type(np.array(deque([1, 2, 3])), npt.NDArray[Any])
 assert_type(np.array(A, dtype=np.int64), npt.NDArray[np.int64])
 assert_type(np.array(A, dtype='c16'), npt.NDArray[Any])
 assert_type(np.array(A, like=A), npt.NDArray[np.float64])
+assert_type(np.array(D), npt.NDArray[np.float64 | np.int64])
+assert_type(np.array(C), npt.NDArray[np.signedinteger[Any]])
+assert_type(np.array(E), npt.NDArray[np.floating[Any]])
```

Comment:
`list[float]` can also contain `int` or `bool`, so we want the returned array to have a scalar type of `float64 | np.int_ | np.bool`.